bears/haml/HAMLLintBear.py

```diff
@@ -0,0 +1,396 @@
+from coalib.results.RESULT_SEVERITY import RESULT_SEVERITY
+from coalib.settings.Setting import typed_list
+from dependency_management.requirements.GemRequirement import GemRequirement
+import yaml
+
+
+@linter(executable='haml-lint',
+        output_format='regex',
+        output_regex=r'(?P<line>\d+) \[(?P<severity>W|E)\] '
+                     r'(?P<message>.*)',
+        severity_map={'E': RESULT_SEVERITY.MAJOR,
+                      'W': RESULT_SEVERITY.NORMAL})
+class HAMLLintBear:
+    """
+    Uses ``haml-lint`` to perform HAML-specific style and lint checks to ensure
+    clean and readable HAML code.
+    """
+    LANGUAGES = {'Haml'}
+    REQUIREMENTS = {GemRequirement('haml_lint', '0.27.0')}
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+    CAN_DETECT = {'Formatting', 'Syntax'}
+    SEE_MORE = 'https://github.com/brigade/haml-lint'
+
+    DEFAULT_IGNORED_COPS = (
+      'Lint/BlockAlignment',
+      'Lint/EndAlignment',
+      'Lint/Void',
+      'Layout/AlignParameters',
+      'Layout/ElseAlignment',
+      'Layout/EndOfLine',
+      'Layout/IndentationWidth',
+      'Layout/TrailingBlankLines',
+      'Layout/TrailingWhitespace',
+      'Metrics/BlockLength',
+      'Metrics/BlockNesting',
+      'Metrics/LineLength',
+      'Naming/FileName',
+      'Style/FrozenStringLiteralComment',
+      'Style/IfUnlessModifier',
+      'Style/Next',
+      'Style/WhileUntilModifier',
+      )
+
+    @staticmethod
+    def generate_config(filename, file,
+                        alignment_tabs: bool=True,
+                        alt_text: bool=True,
+                        class_attribute_with_static_value: bool=True,
+                        classes_before_ids: bool=True,
+                        consecutive_comments: bool=True,
+                        max_consecutive_comments: int=1,
+                        consecutive_silent_scripts: bool=True,
+                        max_consecutive_silent_scripts: int=2,
+                        empty_object_reference: bool=True,
+                        empty_script: bool=True,
+                        final_newline: bool=True,
+                        final_newline_present: bool=True,
+                        html_attributes: bool=True,
+                        id_names: bool=True,
+                        id_names_style: str='lisp_case',
+                        implicit_div: bool=True,
+                        indentation: bool=True,
+                        indentation_character: str='space',
+                        indentation_width: int=2,
+                        inline_styles: bool=True,
+                        instance_variables: bool=True,
+                        instance_variables_file_types: str='partials',
+                        instance_variables_matchers_all: str='.*',
+                        instance_variables_matchers_partials:
+                        str='\A_.*\.haml\z',
```

Comment:
Did you intend to use `r` with the backslashes inside the string?

Reply:
In general, even no backslashes are used, try to use `r` prefixes on strings when those denote regexes :+1: